src/serde/parse_atom.rs

```diff
@@ -9,40 +9,44 @@ const MAX_SINGLE_BYTE: u8 = 0x7f;
 /// decode the length prefix for an atom. Atoms whose value fit in 7 bits
 /// don't have a length prefix, so those should be handled specially and
 /// never passed to this function.
-pub fn decode_size<R: Read>(f: &mut R, initial_b: u8) -> Result<u64> {
+pub fn decode_size_with_offset<R: Read>(f: &mut R, initial_b: u8) -> Result<(u8, u64)> {
     debug_assert!((initial_b & 0x80) != 0);
     if (initial_b & 0x80) == 0 {
         return Err(internal_error());
     }
 
-    let mut bit_count = 0;
+    let mut atom_start_offset = 0;
     let mut bit_mask: u8 = 0x80;
     let mut b = initial_b;
     while b & bit_mask != 0 {
-        bit_count += 1;
+        atom_start_offset += 1;
         b &= 0xff ^ bit_mask;
         bit_mask >>= 1;
     }
     let mut size_blob: Vec<u8> = Vec::new();
-    size_blob.resize(bit_count, 0);
+    size_blob.resize(atom_start_offset, 0);
```

Comment:
> don't have a length prefix, so those

For now, it's just used by `decode_size`. It will be used elsewhere later. I'm just trying to get smaller PRs through for now.

Reply:
Turns out we have to reserve 8 bytes or the fuzzer can crash things.